exporter/elasticsearchexporter/integrationtest/exporter_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package integrationtest
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestExporter(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		restartCollector bool
+		mockESFailure    bool
+	}{
+		{name: "basic"},
+		{name: "es_intermittent_failure", mockESFailure: true},
+		/* Below tests should be enabled after #30792 is fixed
+		{name: "collector_restarts", restartCollector: true},
+		{name: "collector_restart_with_es_intermittent_failure", mockESFailure: true, restartCollector: true},
+		*/
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			runner(t, tc.restartCollector, tc.mockESFailure)
+		})
+	}
+}
+
+func runner(t *testing.T, restartCollector, mockESFailure bool) {
+	t.Helper()
+
+	cfg := loadConfig(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	flushIvl := time.Second
+	mockES := newMockESClient(t, cfg.Debug)
+	collector := newTestCollector(t, cfg, mockES.ServerURL, flushIvl)
+
+	var g errgroup.Group
+	cancelRun := runTestCollectorWithWait(ctx, t, collector, &g)
+
+	if mockESFailure {
+		mockES.SetReturnStatusCode(http.StatusServiceUnavailable)
+		if !restartCollector {
+			// If restartCollector is not set then recover ES service
+			// after a few failed bulk requests otherwise ensure that
+			// both collector and ES are unavailable for the same
+			// duration considering worst case scenario. We use a
+			// multiple of flush interval to determine the downtime for
+			// ES. Using a multiple greater than 1 ensures that there
+			// is enough downtime to make flush/bulk reqests to ES.
+			time.AfterFunc(10*flushIvl, func() {
+				mockES.SetReturnStatusCode(http.StatusOK)
+			})
```

Comment:
Rather than sleeping, should we wait for requests to actually be performed by the mock client?

Reply:
This has also been discussed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31848#discussion_r1532756208

> Rather than sleeping, should we wait for requests to actually be performed by the mock client?

@ycombinator also pointed this out and I think we can do this but my only concern here is that it would bloat the mock ES a bit (we would have to add logic to count the number of bulk requests and then add hooks to react to it). Additionally, I am not sure if it would be any better than what we have right now (discussion in the above-linked thread).

Reply:
Let me try adding the code to see how this comes out.

Reply:
> https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/3364b7515cf15293f06c976b65a957e4d3447cbe

@axw @ycombinator I gave it a shot and it turned out to be okay. Above is the commit for the changes, I have not yet pushed them to this branch. Let me know your thoughts.

Reply:
Looks good to me.  Thanks for iterating on this.

Reply:
++ LGTM

Reply:
Thanks for reviewing, updated the PR with the changes.